ci_framework/roles/ci_nmstate/filter_plugins/ci_nmstate_map_instance_nets_filter.py

```diff
@@ -0,0 +1,167 @@
+        - CI-framework networks dictionary for a given instance.
+        - Each key represents a network.
+        - Each value represents properties of the interface attached to that network.
+      type: dict
+      required: true
+'''
+
+EXAMPLES = '''
+- name: Define data to work on in the examples below
+  vars:
+    instace_nets:
+      default:
+        connection: ci-private-network
+        gw: 192.168.122.1
+        iface: eth1
+        ip: 192.168.122.100/24
+        mac: fa:16:3e:e4:b3:55
+        mtu: 1500
+      internal-api:
+        iface: eth1.20
+        ip: 172.17.0.100/24
+        mac: 52:54:00:eb:ce:f2
+        mtu: '1496'
+        parent_iface: eth1
+        vlan: 20
+  ansible.builtin.set_fact:
+    nmstate_interfaces: >-
+      {{
+        instace_nets | ci_nmstate_map_instance_nets
+      }}
+'''
+
+RETURN = '''
+  _value:
+    description: The result of the query.
+    type: list
+    sample:
+      - description: default
+        ipv4:
+          address:
+            - ip: 192.168.122.100
+              prefix-length: 24
+              enabled: true
+          ipv6:
+              enabled: false
+          mtu: 1500
+          name: eth1
+          state: up
+          type: ethernet
+      -   description: internal-api
+          ipv4:
+              address:
+              -   ip: 172.17.0.100
+                  prefix-length: 24
+              enabled: true
+          ipv6:
+              enabled: false
+          mtu: 1496
+          name: eth1.20
+          state: up
+          type: vlan
+          vlan:
+              base-iface: eth1
+              id: 20
+'''
+
+import ipaddress
+
+
+from ansible.errors import AnsibleFilterTypeError
+
+
+class FilterModule:
+    @staticmethod
+    def __map_instace_ip(ip_interface):
+        ip_field = {
+            "enabled": bool(ip_interface),
+        }
+        if ip_interface:
+            nmstate_addr_entry = {
+                "ip": str(ip_interface.ip),
+                "prefix-length": ip_interface.network.prefixlen,
+            }
+            ip_field["address"] = [nmstate_addr_entry]
+        return ip_field
+
+    @classmethod
+    def __map_intance_ipv4(cls, ci_instance_net):
+        ip_str = ci_instance_net.get("ip4", ci_instance_net.get("ip", None))
+        ip_interface = ipaddress.ip_interface(ip_str) if ip_str else None
+        return {
+            "ipv4": cls.__map_instace_ip(
+                ip_interface if ip_interface and ip_interface.version == 4 else None
+            )
+        }
+
+    @classmethod
+    def __map_intance_ipv6(cls, ci_instance_net):
+        ip_str = ci_instance_net.get("ip6", ci_instance_net.get("ip", None))
+        ip_interface = ipaddress.ip_interface(ip_str) if ip_str else None
+        return {
+            "ipv6": cls.__map_instace_ip(
+                ip_interface if ip_interface and ip_interface.version == 6 else None
+            )
+        }
+
+    @staticmethod
+    def __map_instace_vlan(ci_instance_net):
```

Comment:
typo this should be `__map_instance_vlan`. It's currently missing the second 'n' from instance.